tests/integration/test_utils.py

```diff
@@ -38,7 +38,7 @@ def retry_until_ok(url, func, print_failure):
                     print('Failed to download otel messages. Response code:',
                         response.status_code)
 
-                return False
```

Comment:
When a network exception happened, the running test was marked as successful. This fixes it and allows to retry this step of the test.